src/model_vit.py

```diff
@@ -154,14 +194,42 @@ def predict_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
             [self.B, 768]  # (batch_size, hidden_size)
         )
 
-        # Save embeddings in npy format
+        # Create table to store the embeddings with spatiotemporal metadata
+        unique_epsg_codes = set(int(epsg) for epsg in epsgs)
+        if len(unique_epsg_codes) == 1:  # check that there's only 1 unique EPSG
+            epsg: int = batch["epsg"][0]
+        else:
+            raise NotImplementedError(
+                f"More than 1 EPSG code detected: {unique_epsg_codes}"
+            )
+
+        gdf = gpd.GeoDataFrame(
+            data={
+                "date": gpd.pd.to_datetime(arg=dates, format="%Y-%m-%d").astype(
+                    dtype="date32[day][pyarrow]"
+                ),
+                "embeddings": pa.FixedShapeTensorArray.from_numpy_ndarray(
+                    embeddings_mean.cpu().detach().__array__()
+                ),
```

Comment:
Although we've converted the embedding into a FixedShapeTensorArray here, `pandas`/`geopadas` still interprets this column as an `object` dtype, and this is saved as an object dtype to the parquet file too (see the unit test). Need to see if there's a way to preserve the dtype.

Reply:
Found a way to save this `embeddings` column as a [FixedShapeTensorArray](https://arrow.apache.org/docs/13.0/python/generated/pyarrow.FixedShapeTensorArray.html) dtype instead of an `object` dtype like so:

```suggestion
                "embeddings": gpd.pd.arrays.ArrowExtensionArray(
                    values=pa.FixedShapeTensorArray.from_numpy_ndarray(embeddings)
                ),
```
However, while we can save this `FixedShapeTensorArray` to GeoParquet, loading this embeddings column as a `FixedShapeTensorArray` is challenging, and might involve code that looks like this:
```python
geodataframe: gpd.GeoDataFrame = gpd.read_parquet(
    path="data/embeddings/embeddings_0.gpq",
    schema=pa.schema(
        fields=[
            pa.field(
                name="embeddings",
                type=pa.fixed_shape_tensor(
                    value_type=pa.float32(), shape=[768]
                ),
            ),
            pa.field(name="geometry", type=pa.binary()),
        ]
    ),
)
```
But this technically still results in an `embeddings` column with `object` dtype... Also, QGIS can load this geoparquet file, but would crash when you try to open the attribute table, because it can't handle FixedShapeTensorArray yet. So probably best to keep it in `object` dtype for now.